packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx

```diff
@@ -122,24 +121,23 @@ const ChallengeTeamWizard: React.FunctionComponent<
   const { data: challenge } = useGetEntityChallenge(projectId)
   const participantTeamId = challenge ? challenge.participantTeamId : EMPTY_ID
   const userId = userProfile ? userProfile.ownerId : EMPTY_ID
+
   // Verify that user is a member of the participant team
-  if (participantTeamId && userId) {
-    getMembershipStatus(participantTeamId, userId)
-      .then(challengeTeamMembershipStatus => {
-        if (!challengeTeamMembershipStatus.isMember && accessToken) {
-          addTeamMemberAsAuthenticatedUserOrAdmin(
-            participantTeamId,
-            userId,
-            accessToken,
-          ).catch(error => {
-            setErrorMessage(error.reason)
-          })
-        }
-      })
-      .catch(error => {
+  const { data: challengeTeamMembershipStatus } = useGetMembershipStatus(
+    participantTeamId,
+    userId,
+  )
+  useEffect(() => {
+    if (!challengeTeamMembershipStatus?.isMember && accessToken) {
+      addTeamMemberAsAuthenticatedUserOrAdmin(
+        participantTeamId,
+        userId,
+        accessToken,
+      ).catch(error => {
```

Comment:
Would also recommend invalidating the cache for `useGetMembershipStatus` when this completes, but it's not strictly necessary, since `challengeTeamMembershipStatus` won't change and cause this effect to run multiple times.